`store_dataframes_as_dataset` accepts dicts and list of tuples equivalents

SvoONs opened the pull request:
# Description:

This PR makes `io/eager/store_dataframes_as_dataset` accept `dfs` arguments in the following formats:
`{"data": {"table": df}}, {"table": df}, {"data": [("table", df)]}` and `[("table", df)]`.

Docstrings have not been adapted yet.

- [x] Closes #63 
- [x] Tests added / passed
- [x] Passes `./format_code.sh`
- [x] Changelog entry

fjetter commented on kartothek/io/eager.py:
```diff
@@ -482,7 +482,10 @@ def store_dataframes_as_dataset(
         dataset_uuid = gen_uuid()
 
-    if isinstance(dfs, dict):
+    if isinstance(dfs, list) and isinstance(dfs[0], tuple):
```
I'd prefer this logic to be implemented in `parse_input_to_metapartition` since this is used for all interfaces. You can also document the new input format there

fjetter commented:
@SvoONs what's the state of this PR?

SvoONs commented:
> @SvoONs what's the state of this PR?
I just updated the PR. Moved all parsing logic to `parse_input_to_metapartition`, which now accepts `{"data": {"table": df}}, {"table": df}, {"data": [("table", df)]} and [("table", df)]` formats as `obj` input in accordance to the docs.

fjetter commented on kartothek/io_components/metapartition.py:
```diff
@@ -1615,6 +1615,13 @@ def parse_input_to_metapartition(
         if len(obj) == 0:
             return MetaPartition(label=None, metadata_version=metadata_version)
         first_element = obj[0]
+        if isinstance(first_element, tuple):
```
Please document the change in the Docstrings above.

SvoONs replied:
Added docstring documentation

lr4d commented on kartothek/io_components/metapartition.py:
```diff
@@ -1629,7 +1636,9 @@ def parse_input_to_metapartition(
-        if isinstance(obj["data"], list):
+        if not obj.get("data"):
+            data = obj
+        elif isinstance(obj["data"], list):
```
@fjetter since `indices` and `label` are deprecated, shouldn't we also deprecate this format (`{"data": {"table": df}}`) in favor of `{"table":df}`?

lr4d replied:
what I was going for was raising a deprecation warning

fjetter replied:
Yes, but I'm not sure what the future behaviour will be. Without this a deprecation warning seems premature to me. I don't want to write a deprecation warning like: "Please don't use this anymore but I cannot tell you what the future will bring"

SvoONs commented:
> Changelog entry missing ;)

Entry added

lr4d commented on tests/io_components/test_metapartition.py:
```diff
@@ -1627,3 +1627,16 @@ def test_parse_nested_input_schema_compatible_but_different():
+
+def test_parse_input_schema_formats():
+    df = pd.DataFrame({"B": [pd.Timestamp("2019")]})
+    formats_obj = [
```
Nitpick: you could use `@pytest.mark.parametrize` here

lr4d replied:
Nvm. I see that you're re-using the dataframe

SvoONs replied:
Yes, that's why I stuck with the `for` loop here.

lr4d commented on tests/io/eager/test_write.py:
```diff
@@ -177,6 +177,23 @@ def test_store_dataframes_as_dataset_no_pipeline(metadata_version, store):
+    df1 = pd.DataFrame({"B": [pd.Timestamp("2019")]})
+    df2 = pd.DataFrame({"A": [1.4]})
+    formats = [
+        {"data": {"D": df1, "S": df2}},
```
also, `pytest.mark.parametrize`
